mod_reforged/hooks/entity/tactical/humans/master_archer.nut

```diff
@@ -52,23 +52,42 @@
 
 		if (this.m.Items.hasEmptySlot(::Const.ItemSlot.Body))
 		{
-			local armor = ::MSU.Class.WeightedContainer([
+			this.m.Items.equip(::new(::MSU.Class.WeightedContainer([
 				[1, "scripts/items/armor/thick_tunic"],
 				[1, "scripts/items/armor/padded_surcoat"],
 				[1, "scripts/items/armor/gambeson"]
-			]).roll();
-
-			this.m.Items.equip(::new(armor));
+			]).roll()));
 		}
 
 		if (this.m.Items.hasEmptySlot(::Const.ItemSlot.Head))
 		{
-			local helmet = ::MSU.Class.WeightedContainer([
-				[1, "scripts/items/helmets/hood"],
-				[1, "scripts/items/helmets/hunters_hat"]
-			]).rollChance(33);
+			if (this.m.IsMiniboss)
+			{
+				this.m.Items.equip(::new("scripts/items/helmets/greatsword_hat"));
+			}
+			else
+			{
+				this.m.Items.equip(::new(::MSU.Class.WeightedContainer([
```

Comment:
You are equipping directly with a `rollChance` which has a chance to return null. So you need a null check here.
```squirrel
local item = ::MSU.Class.WeightedContainer({blabla}).rollChance(33);
if (item != null)
{
// equip here
}
```